FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,335 @@
+                             example : ['stdout', 'file', 'server']
+    :params backendOptions: a dictionary of different backend options. 
+                            example: {'FileName': '/tmp/log.txt'}
+    """
+    for stringHandler in desiredBackends:
+      stringHandler = stringHandler.lower()
+      stringHandler = stringHandler.strip()
+
+      # check if the name is correct
+      if stringHandler in self._backendsDict:
+        backend = self._backendsDict[stringHandler]
+
+        # check if the backend is already in the list
+        if backend not in self._backendsList:
+          if backendOptions is not None:
+            backend.setParameters(backendOptions.copy())
+
+          backend.configureHandler()
+          self._logger.addHandler(backend.getHandler())
+          self._backendsList.append(backend)
+          self._updateFormat()
+      else:
+        self._updateFormat()
+        self.warn("%s is not a valid backend name.", stringHandler)
+
+  def setLevel(self, levelName):
+    """
+    Set a level to the logger.
+    :params levelName: string representing the level to give to the logger
+    :return: boolean representing if the setting is done or not
+    """
+    result = False
+    levelName = levelName.upper()
+    if levelName in LogLevels.getLevelNames():
+      self._logger.setLevel(LogLevels.getLevelValue(levelName))
+      result = True
+    return result
+
+  def getLevel(self):
+    """
+    :return: the name of the level
+    """
+    return LogLevels.getLevel(self._logger.getEffectiveLevel())
+
+  def shown(self, levelName):
+    """
+    Determine if messages with a certain level will be displayed or not.
+    :params levelName: string representing the level to analyse
+    :return: boolean which give the answer
+    """
+    result = False
+    levelName = levelName.upper()
+    if levelName in LogLevels.getLevelNames():
+      result = self._logger.isEnabledFor(LogLevels.getLevelValue(levelName))
+    return result
+
+  @staticmethod
+  def getName():
+    """
+    :return: "system name/component name"
+    """
+    return gLogging._componentName
+
+  def getSubName(self):
+    """
+    :return: the name of the logger
+    """
+    name = self._logger.name
+    name = name.replace("root", "")
+
+    if name != "":
+      names = name.split('.')
+      name = names[-1]
+
+    return name
+
+  def getDisplayOptions(self):
+    """
+    :return: a copy of the dictionary of the display options and their values
+    """
+    return self._options.copy()
+
+  def _setChildrensDisplayOptions(self, optionName, options):
+    """
+    Set the display options of the childrens if they are not modified by the user
+    """
+    if not self._optionsmodified[optionName]:
+      self._options = options.copy()
+      self._updateFormat()
+    for child in self._childrens:
+      child._setChildrensDisplayOptions(optionName, options)
+
+  def getAllPossibleLevels(self):
```

Comment:
getAllPossibleLevels() ? Just in one case in Interfaces/API/Job in the setLogLevel() method